docs/src/proposals/scrambled-tx.md

```diff
@@ -0,0 +1,335 @@
+scrambled because scrambling might not be desirable for all of situations (HD
+wallets and program deploy transactions).
+
+### Scrambling preparation by clients
+
+Clients need to prepare valid scrambling challenge secrets. By definition, they
+have no choice but to compute the recursive sha256, starting from
+locally-available secure random as seed. However, the challenge secrets are
+designed to be pre-computed before the actual transaction construction/signing.
+So, they can prepare those challenge secrets in background and idling manner for
+later use.
+
+### 2 step voting and voting lock-out
+
+pre-vote and vote are expected to be coalesced to avoid doubling the consensus
+communication overhead. This means a single voting transaction will include
+instructions to pre-vote on block N and to vote on block N-1. This shouldn't
+badly affect latency because block time is still 400ms frequency. Also, the
+pre-vote instruction must contain timestamp.
+
+pre-vote and vote on block N must occur in order. And voting rules including
+lockout are generalized as if any vote comprises of the 2 steps. So fork-choice
+and related subsystem isn't changed significantly.
+
+### Latency analysis
+
+Scrambling will inevitably increases latency. However this should be tolerable
+in trade for the MEV immunity.
+
+Descrambling difficulty can be targeted to be 2x of current opt-conf duration as
+a very crude approach. The 2x is needed as a safe buffer (TODO: needs proper
+statistical analysis) for variation of sha256 throughput across hardware.
+
+However this is quite inefficient and too naive.
+
+Firstly, the current opt-conf duration is encompassing all the timings,
+including block propagation, sig/entry verify, transaction replay and vote
+accumulation. The new pre-vote opt-conf duration should be a small portion of
+it, namely the block propagation and vote accumulation.
+
+We can also exclude the vote accumulation duration because the block propagation
+duration is the only duration which needs to remain scrambled. We can
+stake-weightedly derive the duration of block propagation from stake-weighted
+timestamps in past pre-votes (could be hard-coded initially) on epoch basis.
+
+Lastly, block propagation for any given slot should finish within a slot or
+two, otherwise cluster isn't healthy in the first place, causing unrecoverable
+desync acorss significant number of staked nodes.
+
+Thus, approximately 4 slots (~1.6s) will be scrambling target difficulty. This
+is major component of induced latency. In addition to this, scrambled keys must
+also propagated via turbine while being pipelined. This will also add ~2 slots
+latency, totaling ~6 slots (~2.4s).
+
+As before, latencies from sig/entry verify, replay, and vote accumulation still
+exist. However scrambling by itself won't cause pipeline stalls. Thus there
+should be no more latency sources, other than described in the previous
+paragraph.
+
+### Attack scenarios
```

Comment:
hmm, i just realized some bait-and-fish attack might be of concern:

trade 1000000 TOKEN on amm is descrabmled => other mev bot tries to win the chance 

however there's the (induced) latency here. the the attacker can actually submit another transaction like the reverse trade between the delay.

in other words, this is exploiting the induced feedback loop delay.. hmm, i think the solution would be proper slippage tolerance in the mev's txes..

Reply:
also note that the same problems exists already regardless of this proposal or not.